### PR TITLE
fix: modify function signatures of GeneratedModel

### PR DESCRIPTION
**Pull request by kimusaku**

### CODE REVIEW

File: sapientml/util/json_util.py [hunk 1]
```diff
@@ -71,7 +71,7 @@ def encode(self, obj):
         """
 
         def replace_nan(o):
-            if isinstance(o, float) and math.isnan(o):
+            if isinstance(obj, float) and math.isnan(obj):
```
[the remark this answers is not in the document]
Comment (kimusaku) [reply sapientml/util/json_util.py]: Thanks. I fixed it by https://github.com/sapientml/sapientml/pull/55/commits/ee9f266f8215ca6bbdf84086690f0ec5583f4d80

File: sapientml/util/json_util.py [hunk 2]
```diff
@@ -96,15 +96,15 @@ def default(self, obj):
...
 
         """
-        if obj == "nan":
+        if obj == "null":
             return np.nan
-        if obj == "inf":
+        if obj == "Infinity":
             return np.inf
-        if obj == "-inf":
+        if obj == "-Infinity":
```
[the remark this answers is not in the document]
Comment (kimusaku) [reply sapientml/util/json_util.py]: Thanks. I did not understand where JSONDecoder should be used in. I reverted the code.

[the remark this answers is not in the document]
Comment (kimusaku) [reply sapientml/util/json_util.py]: I correct the implementation of JSONDecoder. Please confirm.

**Review by AkiraUra — APPROVED** [review]
Thank you!

**Pull request merged**